content/tutorials/custom-zk-chain/20.customizing-your-chain.md

```diff
@@ -0,0 +1,393 @@
+
+import dotenv from "dotenv";
+dotenv.config();
+
+const config: HardhatUserConfig = {
+  solidity: "0.8.20",
+  networks: {
+        localRethNode: {
+          url: "http://localhost:8545",
+          accounts: [process.env.WALLET_PRIVATE_KEY as any],
+        },
+      },
+};
+
+export default config;
+```
+
+Now hardhat is configured to connect to our local reth node (the L1) running at local port `8545`.
+For the inital release of the ZK stack, we have to deploy the token to the L1 and bridge to our chain later.
+However, in the future this may change.
+
+Next, create a `.env` file with:
+
+```bash
+touch .env
+```
+
+Add the `WALLET_PRIVATE_KEY` environment variable with the private key of the rich wallet we've been using
+so that we can deploy using a pre-funded address:
+
+```env
+WALLET_PRIVATE_KEY=0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110
+```
+
+### Deploying an ERC20 Contract
+
+Now that we've configured hardhat and the deployer wallet, let's add the contract and deploy script.
+Rename the example generated contract file to `CustomBaseToken.sol`:
+
+```bash
+mv contracts/Lock.sol contracts/CustomBaseToken.sol
+```
+
+Open the `CustomBaseToken.sol` file and replace the example contract with the ERC20 contract below:
+
+```solidity
+// SPDX-License-Identifier: Unlicensed
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+
+contract CustomBaseToken is ERC20, Ownable, ERC20Burnable {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) Ownable(msg.sender) {
+        _mint(msg.sender, 100 * 10 ** decimals());
+    }
+
+    function mint(address to, uint256 amount) public onlyOwner {
+        _mint(to, amount);
+    }
+}
+```
+
+Next, let's update the ignition module to deploy the ERC20 contract.
+Rename the module file with the command below:
+
+```bash
+mv ignition/modules/Lock.ts ignition/modules/CustomBaseToken.ts
+```
+
+Then, replace the module file with the code below:
+
+```ts
+import { buildModule } from "@nomicfoundation/hardhat-ignition/modules";
+
+const CustomBaseTokenModule = buildModule("CustomBaseTokenModule", (m) => {
+  const tokenName = m.getParameter("name", "ZK Base Token");
+  const tokenSymbol = m.getParameter("symbol", "ZKBT");
+
+  const baseToken = m.contract("CustomBaseToken", [tokenName, tokenSymbol]);
+
+  return { baseToken };
+});
+
+export default CustomBaseTokenModule;
+```
+
+To run the module and deploy the token contract, run:
+
+```bash
+npx hardhat ignition deploy ./ignition/modules/CustomBaseToken.ts --network localRethNode
+```
+
+Select `y` to confirm to deploy the contract to the `localRethNode`.
+After deploying, the token contract address should be logged in your console.
+
+Let's verify that the tokens were minted to the deployer address using the Foundry `cast` CLI:
```

Comment:
When were tokens minted for the deployer address? Is that part of the ignition module?

Reply:
yes, the ignition module is deploying the token

Reply:
ah, I glossed over the token contract